homeassistant/components/tuya/light.py

```diff
@@ -64,6 +86,47 @@ def __init__(self, tuya, platform):
         """Init Tuya light device."""
         super().__init__(tuya, platform)
         self.entity_id = ENTITY_ID_FORMAT.format(tuya.object_id())
+        self._min_kelvin = tuya.max_color_temp()
+        self._max_kelvin = tuya.min_color_temp()
+
+    @callback
+    def _load_config(self):
+        """Set device config parameter."""
+        config = self._get_device_config()
+        if not config:
+            return
+
+        # support color config
+        supp_color = config.get(CONF_SUPPORT_COLOR, False)
+        if supp_color:
+            self._tuya.force_support_color()
+        # brightness range config
+        self._tuya.brightness_white_range = BRIGHTNESS_MODES.get(
+            config.get(CONF_BRIGHTNESS_RANGE_MODE, 0),
+            TUYA_BRIGHTNESS_RANGE0,
+        )
+        # color set temp range
+        min_tuya = self._tuya.max_color_temp()
+        min_kelvin = config.get(CONF_MIN_KELVIN, min_tuya)
+        max_tuya = self._tuya.min_color_temp()
+        max_kelvin = config.get(CONF_MAX_KELVIN, max_tuya)
+        self._min_kelvin = min(max(min_kelvin, min_tuya), max_tuya)
+        self._max_kelvin = min(max(max_kelvin, self._min_kelvin), max_tuya)
+        # color shown temp range
+        max_color_temp = max(
+            config.get(CONF_TUYA_MAX_COLTEMP, DEFAULT_TUYA_MAX_COLTEMP),
+            DEFAULT_TUYA_MAX_COLTEMP,
+        )
+        self._tuya.color_temp_range = (1000, max_color_temp)
+
+    async def async_added_to_hass(self):
+        """Set config parameter when add to hass."""
+        await super().async_added_to_hass()
+        self._load_config()
```

Comment:
```suggestion
        self._process_config()
```

Can we avoid calling this `load` since its in `async` and as it implies we are doing I/O

Reply:
Renamed also in `climate.py`